src/dvc_data/hashfile/db/local.py

```diff
@@ -101,18 +102,13 @@ def _unprotect_file(self, path):
 
         os.chmod(path, self._file_mode)
 
-    def _unprotect_dir(self, path):
-        for fname in self.fs.find(path):
-            self._unprotect_file(fname)
-
-    def unprotect(self, path):
+    def unprotect(self, path, callback=DEFAULT_CALLBACK):
         if not os.path.exists(path):
             raise ObjectDBError(f"can't unprotect non-existing data '{path}'")
 
-        if os.path.isdir(path):
-            self._unprotect_dir(path)
-        else:
-            self._unprotect_file(path)
+        files = self.fs.find(path) if os.path.isdir(path) else [path]
```

Comment:
Should we do a `set_size` here? i.e that would mean creating a list of files in memory?